docs/development.rst

```diff
@@ -83,3 +84,59 @@ install the bleeding-edge package directly from the tskit github repository requ
 to specify ``subdirectory=python``:
 
 ``python3 -m pip install git+https://github.com/tskit-dev/tskit.git#subdirectory=python``
+
+***********************************************
+Best Practices for adding to the tskit codebase
+***********************************************
+
+The following is a rough guide of best practices for contributing a function to the
+tskit codebase.
+We follow the example of the `kc_distance function` 
+
+0.  Open an `issue <http://github.com/tskit-dev/tskit/issues>`_ with your proposed
+    functionality. If consensus is reached that your proposed addition shoudl be
+    added to the codebase, proceed! 
+1.  Create a new branch on your fork of tskit-dev (see :ref:`sec_development_quickstart`
+    above). Open a `pull request <http://github.com/tskit-dev/tskit/pulls>`_ on GitHub,
+    with an initial description of your planned addition.
+2.  Write your function in Python: In python/tests, find the test module that 
+    pertains to the functionality you wish to add. For instance, the kc_distance
+    metric was added to 
+    `test_topology.py <https://github.com/tskit-dev/tskit/blob/master/python/tests/test_topology.py>`_.
+    Add a python version of your function here.
+3.  Create a new class in this module to write unit tests for your function. In addition
+    to making sure that your function is correct, make sure it fails on inappropriate inputs.
+    This can often require judgement. For instance, kc_distance fails on a tree with multiple
+    roots, but allows users to input parameter values that are nonsensical, but that don't 
+    break functionality. 
+4.  Write your function in C: check out the :ref:`sec_c_api` for guidance. There
+    are also many examples in the 
+    `c directory <https://github.com/tskit-dev/tskit/tree/master/c/tskit>`_.
+    Your function will probably go in 
+    `trees.c <https://github.com/tskit-dev/tskit/blob/master/c/tskit/trees.c>`_.
+5.  Write a few tests for your function in C: again, write your tests in  
+    `tskit/c/tests/test_tree.c <https://github.com/tskit-dev/tskit/blob/master/c/tests/test_trees.c>`_.
+    The key here is code coverage, you don't need to worry as much about covering every
+    corner case, as we will proceed to link this function to the Python tests you
+    wrote earlier.
+6.  Create a low-level definition of your function using Python's C API: This will
+    go in `tskit/python/_tskit_module.c 
+    <https://github.com/tskit-dev/tskit/blob/master/python/_tskitmodule.c>`_.
+7.  Test your low-level implementation in `tskit/python/tests/test_lowlevel.py
+    <https://github.com/tskit-dev/tskit/blob/master/python/tests/test_lowlevel.py>`_.
```

Comment:
, concentrating particularly on the *interface* (does the function behave correctly on malformed inputs), etc.